collector/powersupplyclass.go

```diff
@@ -0,0 +1,183 @@
+	powerSupplyClass, err := getPowerSupplyClassInfo(c.ignoredPattern)
+	if err != nil {
+		return fmt.Errorf("could not get power_supply class info: %s", err)
+	}
+	for _, powerSupply := range powerSupplyClass {
+
+		for name, value := range map[string]*int64{
+			"authentic":                   powerSupply.Authentic,
+			"calibrate":                   powerSupply.Calibrate,
+			"capacity":                    powerSupply.Capacity,
+			"capacity_alert_max":          powerSupply.CapacityAlertMax,
+			"capacity_alert_min":          powerSupply.CapacityAlertMin,
+			"charge_control_limit":        powerSupply.ChargeControlLimit,
+			"charge_control_limit_max":    powerSupply.ChargeControlLimitMax,
+			"charge_counter":              powerSupply.ChargeCounter,
+			"charge_empty":                powerSupply.ChargeEmpty,
+			"charge_empty_design":         powerSupply.ChargeEmptyDesign,
+			"charge_full":                 powerSupply.ChargeFull,
+			"charge_full_design":          powerSupply.ChargeFullDesign,
+			"charge_now":                  powerSupply.ChargeNow,
+			"charge_term_current":         powerSupply.ChargeTermCurrent,
+			"constant_charge_current":     powerSupply.ConstantChargeCurrent,
+			"constant_charge_current_max": powerSupply.ConstantChargeCurrentMax,
+			"constant_charge_voltage":     powerSupply.ConstantChargeVoltage,
+			"constant_charge_voltage_max": powerSupply.ConstantChargeVoltageMax,
+			"current_avg":                 powerSupply.CurrentAvg,
+			"current_boot":                powerSupply.CurrentBoot,
+			"current_max":                 powerSupply.CurrentMax,
+			"current_now":                 powerSupply.CurrentNow,
+			"cyclecount":                  powerSupply.CycleCount,
+			"energy_avg":                  powerSupply.EnergyAvg,
+			"energy_empty":                powerSupply.EnergyEmpty,
+			"energy_empty_design":         powerSupply.EnergyEmptyDesign,
+			"energy_full":                 powerSupply.EnergyFull,
+			"energy_full_design":          powerSupply.EnergyFullDesign,
+			"energy_now":                  powerSupply.EnergyNow,
+			"input_current_limit":         powerSupply.InputCurrentLimit,
+			"online":                      powerSupply.Online,
+			"power_avg":                   powerSupply.PowerAvg,
+			"power_now":                   powerSupply.PowerNow,
+			"precharge_current":           powerSupply.PrechargeCurrent,
+			"present":                     powerSupply.Present,
+			"temp":                        powerSupply.Temp,
+			"temp_alert_max":              powerSupply.TempAlertMax,
+			"temp_alert_min":              powerSupply.TempAlertMin,
+			"temp_ambient":                powerSupply.TempAmbient,
+			"temp_ambient_max":            powerSupply.TempAmbientMax,
+			"temp_ambient_min":            powerSupply.TempAmbientMin,
+			"temp_max":                    powerSupply.TempMax,
+			"temp_min":                    powerSupply.TempMin,
+			"time_to_empty_now":           powerSupply.TimeToEmptyNow,
```

Comment:
This should probably be `time_to_empty_seconds`.

Reply:
Done